Services/CAS/classes/class.ilAuthProviderCAS.php

```diff
@@ -38,29 +67,54 @@ protected function getSettings(): ilCASSettings
     public function doAuthentication(ilAuthStatus $status): bool
     {
         $this->getLogger()->debug('Starting cas authentication attempt... ');
+        $baseUrl = $this->getIliasBaseUrl();
 
         try {
-            phpCAS::setDebug(false);
-            phpCAS::setVerbose(true);
+            // If you need a logger, uncomment this, the use statements
+            // and variables above and remove   phpCAS::setLogger(null);
+            // from below.
+            // The logger essentially gives you a trace through the
+            // entire CAS-call-stack.
+            //
+            // $this->logPath = tempnam(sys_get_temp_dir(), 'phpCAS');
+            // $this->logger = new Logger('name');
+            // $handler = new StreamHandler($this->logPath);
+            // $format = "%message%\n";
+            // $formatter = new LineFormatter($format);
+            // $handler->setFormatter($formatter);
+            // $this->logger->pushHandler($handler);
+            // phpCAS::setLogger($this->logger);
+
+            phpCAS::setLogger(null);
+            // Caution: If you set this to "true", there might be output
+            // and the redirect won't work and you get an ILIAS Whoopsy
+            // Though, you may need to for debugging other issues.
+            phpCAS::setVerbose(false);
+            $this->getLogger()->debug('Create client... ');
             phpCAS::client(
                 CAS_VERSION_2_0,
                 $this->getSettings()->getServer(),
                 $this->getSettings()->getPort(),
-                $this->getSettings()->getUri()
+                $this->getSettings()->getUri(),
+                $baseUrl
```

Comment:
Can't we use `ilUtil::_getHttpPath()` instead?

Reply:
> if it is an officially supported feature of ILIAS to work under multiple domain names

It's not.

I suggest using the above method to avoid introducing another “truth” regarding the “HTTP path”.